src/ffi.rs

```diff
@@ -766,6 +766,26 @@ pub extern fn quiche_conn_trace_id(
     *out_len = trace_id.len();
 }
 
+#[no_mangle]
+pub extern fn quiche_conn_source_id(
+    conn: &mut Connection, out: &mut *const u8, out_len: &mut size_t,
+) {
+    let id = conn.source_id().to_vec();
```

Comment:
This value is dropped once the function returns, so I think this would end up returning a pointer to freed memory to the application.

Instead of doing `to_vec()` (which allocates a new value withing this function), I think `as_ref()` would be better, so we would use a pointer to the actual value held by the connection object (which currently has the same lifetime as the connection itself), rather than a newly allocated but short-lived value.

But probably worth confirming with valgrind in an application that uses these new functions.

Reply:
Sounds about right ... let me change it